read-fonts/src/tables/glyf.rs

```diff
@@ -945,6 +958,52 @@ mod tests {
         );
     }
 
+    #[test]
+    fn simple_glyph_overlapping_contour_flag() {
+        let font = FontRef::new(font_test_data::VAZIRMATN_VAR).unwrap();
+        let loca = font.loca(None).unwrap();
+        let glyf = font.glyf().unwrap();
+        let glyph_count = font.maxp().unwrap().num_glyphs();
+        for gid in 0..glyph_count {
+            let glyph = match loca.get_glyf(GlyphId::new(gid), &glyf) {
+                Ok(Some(Glyph::Simple(glyph))) => glyph,
+                _ => continue,
+            };
+            if gid == 3 {
+                // Only GID 3 has the overlap bit set
+                assert!(glyph.has_overlapping_contours())
+            } else {
+                assert!(!glyph.has_overlapping_contours())
+            }
+        }
+    }
+
+    #[test]
+    fn composite_overlapping_contour_flag() {
```

Comment:
The loading of test data, num glyphs, etc is repetitive noise; perhaps a test helper that lets you easily iterate over (gid, &Glyph) would help tidy?

Reply:
The refactoring part of my brain seems to turn off when writing tests. Good call, ty!